terraform/deployments/create-or-update/main.tf

```diff
@@ -42,6 +42,38 @@ resource "azurerm_nginx_deployment" "example" {
   tags = var.tags
 }
 
+resource "azurerm_nginx_configuration" "aniruddh-example-config" {
+  nginx_deployment_id = azurerm_nginx_deployment.aniruddh-example.id
```

Comment:
this should be `azurerm_nginx_deployment.example.id`

Reply:
Yes, changed.